### ISSUE TITLE
Reupload to PyPI to take advantage of pyproject.toml

### ISSUE THREAD

**Issue by alugowski** [scikit-learn-contrib/hdbscan#233]
This commit was recently merged: https://github.com/scikit-learn-contrib/hdbscan/commit/7a1f7738e9b095fa2454687f2afc09aa6b7581d6

It adds a `pyproject.toml` which means `pip install hdbscan` will work even if you don't have Cython installed yet. This makes it possible to use hdbscan in `requirements.txt` files and have it work as expected. It also means it's possible to use hdbscan in places that only allow one `pip install` statement, such as Google AppEngine.

This only requires a minor version bump and a reupload, nothing else.

Thank you for your work!

**Comment by lmcinnes** [comment]
Done. Thanks for the nudge. Wheels should be coming too soon.

**Comment by alugowski** [comment]
Thank you!

I tested the new version from PyPI and it still shows the Cython warning. Upon further research it looks like `pyproject.toml` needs to be included in the tarball to have an effect, and 0.8.16 doesn't do that.

I created a PR that adds `pyproject.toml` to `MANIFEST.in`: https://github.com/scikit-learn-contrib/hdbscan/pull/234
Now when I build the tarball locally and install it with pip now the right thing happens. So I think that's all that's needed to make it work properly from PyPI. 

Note that this whole thing requires `pip` 10 or newer to work. Pip 18 comes with Python 3.7, but older Pythons bundle a pip that's too old.

Still, this should significantly simplify installation instructions:

```
pip install --upgrade pip  # optional
pip install hdbscan
```
even in a clean virtualenv.

**Comment by lmcinnes** [comment]
Thanks. I am really not great at the whole packaging thin, so I really
appreciate all the assistance that people have provided.

**Comment by alugowski** [comment]
Indeed. The Python distribution system is improving but it's a complicated mess.

**Comment by alugowski** [comment]
Thank you!

The version in PyPI now downloads Cython properly, so this issue appears resolved.

`pip install hdbscan` in a clean virtualenv doesn't work yet, because `scipy` isn't listed as a dependency. Of course today a user can simply install scipy separately and everything will work.

In the interests of completeness and to reduce user error, I created a PR that fixes any missing dependencies: https://github.com/scikit-learn-contrib/hdbscan/pull/236

Since this should resolve many outstanding installation issues, I also simplified and clarified installation instructions: https://github.com/scikit-learn-contrib/hdbscan/pull/237
